url41e0dd6de9_jakaldama_pipeline_simulator_tgz-pJ8-PipelineSimulatorJ8/src/PipelineSimulator.java

```diff
@@ -0,0 +1,590 @@
+		
+		System.out.println();
+		System.out.println("Register File: ");
+		for (int i = 0; i < regs.length; i++) {
+			System.out.printf("%X",regs[i]);
+			System.out.print(" |");
+		}
+		System.out.println();
+		System.out.println();
+		//System.out.println("Main Memory: ");
+		//for (int i = 0; i < mainMemory.length; i++) {
+		//	System.out.print(mainMemory[i] + "|");
+		//}
+		
+		System.out.println();
+		System.out.println();
+		}
+	
+	public static void Copy_write_to_read() {
+		//copies everything from write register to read registers
+
+		IFID.setReadInstruction(IFID.getWriteInstruction());
+		
+		IDEX.setReadInstr(IDEX.getWriteInstr());
+		IDEX.setReadALUOp(IDEX.getWriteALUOp());
+		IDEX.setReadALUSrc(IDEX.getWriteALUSrc());
+		IDEX.setReadALUcontrol(IDEX.getWriteALUcontrol());
+		IDEX.setReadRegWrite(IDEX.getWriteRegWrite());
+		IDEX.setReadMemRead(IDEX.getWriteMemRead());
+		IDEX.setReadMemtoReg(IDEX.getWriteMemtoReg());
+		IDEX.setReadMemWrite(IDEX.getWriteMemWrite());
+		IDEX.setReadRegDst(IDEX.getWriteRegDst());
+		IDEX.setReadFunct(IDEX.getWriteFunct());
+		IDEX.setReadOffset(IDEX.getWriteOffset());
+		IDEX.setReadOpCode(IDEX.getWriteOpCode());
+		IDEX.setReadRegD(IDEX.getWriteRegD());
+		IDEX.setReadRegS(IDEX.getWriteRegS());
+		IDEX.setReadRegT(IDEX.getWriteRegT());
+		IDEX.setReadReadData1(IDEX.getWriteReadData1());
+		IDEX.setReadReadData2(IDEX.getWriteReadData2());
+
+		
+		EXMEM.setReadInstr(EXMEM.getWriteInstr());
+		EXMEM.setReadMemRead(EXMEM.getWriteMemRead());
+		EXMEM.setReadMemtoReg(EXMEM.getWriteMemtoReg());
+		EXMEM.setReadMemWrite(EXMEM.getWriteMemWrite());
+		EXMEM.setReadRegD(EXMEM.getWriteRegD());
+		EXMEM.setReadRegS(EXMEM.getWriteRegS());
+		EXMEM.setReadRegT(EXMEM.getWriteRegT());
+		EXMEM.setReadOffset(EXMEM.getWriteOffset());
+		EXMEM.setReadRegWrite(EXMEM.getWriteRegWrite());
+		EXMEM.setReadALUresult(EXMEM.getWriteALUresult());
+		EXMEM.setReadWriteRegister(EXMEM.getWriteWriteRegister());
+		EXMEM.setReadReadData1(EXMEM.getWriteReadData1());
+		EXMEM.setReadReadData2(EXMEM.getWriteReadData2());
+
+		MEMWB.setReadData(MEMWB.getWriteData());
+		MEMWB.setReadInstr(MEMWB.getWriteInstr());
+		MEMWB.setReadMemtoReg(MEMWB.getWriteMemtoReg());
+		MEMWB.setReadRegWrite(MEMWB.getWriteRegWrite());
+		MEMWB.setReadALUresult(MEMWB.getWriteALUresult());
+		MEMWB.setReadOffset(MEMWB.getWriteOffset());
+		MEMWB.setReadRegD(MEMWB.getWriteRegD());
+		MEMWB.setReadRegS(MEMWB.getWriteRegS());
+		MEMWB.setReadRegT(MEMWB.getWriteRegT());
+		MEMWB.setReadWriteRegister(MEMWB.getWriteWriteRegister());
+
+	}
+	
+	public static boolean rType(int i) {
+		if (((i >>> 26) & 0x3f) == 0) {
+			return true;
+		}
+		return false;
+	}
+	
+	public static boolean load(int i) {
+		if (((i >>> 26) & 0x3f) == 0x20) {
+			return true;
+		}
+		return false;
+	}
+	
+	public static boolean store(int i) {
+		if (((i >>> 26) & 0x3f) == 0x28) {
+			return true;
+		}
+		return false;
+	}
+	
+	public static void setControlSignals() {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 15. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test.

We recommend that you simplify this method or break it into multiple methods. For example, consider extracting the code block on lines 486-497 into a separate method.